File: 2025-07-July-GFG-POTD/July 24 - Last Moment Before All Ants Fall Out/Solution.py
class Solution:
    def getLastMoment(self, n: int, left: list[int], right: list[int]) -> int:
        max_time = 0

        # Time for ants moving left: distance to 0
        for pos in left:
            max_time = max(max_time, pos)

        # Time for ants moving right: distance to n
        for pos in right:
            max_time = max(max_time, n - pos)

        return max_time

File: 2025-07-July-GFG-POTD/July 24 - Last Moment Before All Ants Fall Out/Solution.java
class Solution {
    public int getLastMoment(int n, int left[], int right[]) {
        int maxTime = 0;
        
        // Time for ants moving left: distance to 0
        for (int pos : left) {
            maxTime = Math.max(maxTime, pos);
        }
        
        // Time for ants moving right: distance to n
        for (int pos : right) {
            maxTime = Math.max(maxTime, n - pos);
        }
        
        return maxTime;
    }
}
